cupy/cuda/memory.pyx

```diff
@@ -188,6 +199,13 @@ cdef class UnownedMemory(BaseMemory):
             runtime._ensure_context()
             ptr_attrs = runtime.pointerGetAttributes(ptr)
             device_id = ptr_attrs.device
+            if device_id == runtime.cudaCpuDeviceId:
+                # this happens with SystemMemory...
+                device_id = device.get_device_id()
+            # CUDA doesn't track memory allocated through the system malloc
+            if device_id == runtime.cudaInvalidDeviceId and _is_ump_enabled:
```

Comment:
@emcastillo I am wondering if we can just remove `_is_ump_enabled`. It seems I did so too in all the `prefetch` changes I did in this file. Thoughts?

For the record, that `cudaInvalidDeviceId` is raised is likely related to nvbugs 4300350.

Reply:
I am ok with removing it. I need to test the case where UMP is not enabled and we pass a garbage pointer. If the execution fails before reaching this point I assume it is safe to do so.